src/app/login/page.tsx

```diff
@@ -0,0 +1,164 @@
+"use client";
+import Button from "@/components/Button";
+import Image from "next/image";
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
+import { useRouter } from "next/navigation";
+import { useEffect, useState } from "react";
+import { useStore } from "../store";
+const toastr = require("toastr");
+
+const Login = () => {
+	/* Hooks ----------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	const router = useRouter();
+	const supabase = createClientComponentClient();
+	const { user, loginUser, signupUser } = useStore((store) => store);
+
+	const [showLogin, setShowLogin] = useState(true);
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [confirmPassword, setConfirmPassword] = useState("");
+	const [username, setUsername] = useState("");
+
+	/* Effects ----------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	useEffect(() => {
+		if (user) {
+			router.push("/");
+		}
+	}, [user]);
+
+	/* Handlers ----------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	const handleLogin = async () => {
+		if (!email || !password) return;
+
+		const { data, error } = await supabase.auth.signInWithPassword({
+			email,
+			password,
+		});
+
+		if (error) {
+			toastr.error(error.message);
+			return;
+		} else {
+			loginUser(data?.user?.id, () => {
+				router.push("/");
+			});
+		}
+	};
+
+	const handleSignUp = async () => {
+		if (!email || !password || !confirmPassword) return;
+		if (password !== confirmPassword) {
+			toastr.error("Passwords do not match");
+			return;
+		}
+
+		signupUser({ email, password, username }, (id) => {
+			loginUser(id, () => {
+				router.push("/");
+			});
+		});
+	};
```

Comment:
It'd be sweet if we could transfer all this logic into a hook! 😄

Reply:
Want to meet with you to see what you mean by this

Reply:
Yeah man let's do it. Just hit me up in the slack channel when you want to chat.